examples/bigquery-audit-log/bigquery_audit_log.sql

```diff
@@ -0,0 +1,187 @@
+  WHERE
+    protopayload_auditlog.serviceName = 'bigquery.googleapis.com'
+    AND protopayload_auditlog.methodName = 'jobservice.jobcompleted'
+    AND protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.eventName
+    IN ( 'table_copy_job_completed',
+    'query_job_completed',
+    'extract_job_completed',
+    'load_job_completed')
+    AND DATE_DIFF( CURRENT_DATE(), DATE(protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.startTime), month) <= 12
+)
+
+
+/* Query the audit */
+
+SELECT
+  principalEmail,
+  callerIp,
+  serviceName,
+  methodName,
+  eventName,
+  projectId,
+  jobId,
+  errorCode,
+  errorMessage,
+  STRUCT(
+    EXTRACT(MINUTE FROM startTime) AS minuteOfDay,
+    EXTRACT(HOUR FROM startTime) AS hourOfDay,
+    EXTRACT(DAYOFWEEK FROM startTime) - 1 AS dayOfWeek,
+    EXTRACT(DAYOFYEAR FROM startTime) AS dayOfYear,
+    EXTRACT(WEEK FROM startTime) AS week,
+    EXTRACT(MONTH FROM startTime) AS month,
+    EXTRACT(QUARTER FROM startTime) AS quarter,
+    EXTRACT(YEAR FROM startTime) AS year
+  ) as date,
+  createTime,
+  startTime,
+  endTime,
+  runtimeMs,
+  TRUNC(runtimeMs / 1000.0, 2) AS runtimeSecs,
+  tableCopy, /* Querying data specific to Copy operation */
+  1 as numCopies, /* Querying data specific to Copy operation */
+  /* Start: Querying data specific to Load operation */
+    totalLoadOutputBytes,
+  (totalLoadOutputBytes / 1000000000) AS totalLoadOutputGigabytes,
+  (totalLoadOutputBytes / 1000000000) / 1000 AS totalLoadOutputTerabytes,
+  STRUCT(
+    load.sourceUris,
+    STRUCT(
+      load.destinationTable.projectId,
+      load.destinationTable.datasetId,
+      load.destinationTable.tableId,
+      CONCAT(load.destinationTable.datasetId, '.', load.destinationTable.tableId)
+      AS relativePath,
+      CONCAT(load.destinationTable.projectId, '.', load.destinationTable.datasetId,
+      '.', load.destinationTable.tableId)
+      AS absolutePath
+    ) AS destinationTable,
+    load.createDisposition,
+    load.writeDisposition,
+    load.schemaJson
+  ) AS load,
+  1 AS numLoads,
+ /* End: Querying data specific to Load operation */
+ /* Start: Querying data specific to Extract operation */
+ REGEXP_CONTAINS(jobId, 'beam') AS isBeamJob,
+  STRUCT(
+    `extract`.destinationUris,
+    STRUCT(
+      `extract`.sourceTable.projectId,
+      `extract`.sourceTable.datasetId,
+      `extract`.sourceTable.tableId,
+      CONCAT(`extract`.sourceTable.datasetId, '.', `extract`.sourceTable.tableId)
+      AS relativeTableRef,
+      CONCAT(`extract`.sourceTable.projectId, '.', `extract`.sourceTable.datasetId,
+      '.', `extract`.sourceTable.tableId)
+      AS absoluteTableRef
+    ) AS sourceTable
+  ) AS `extract`,
+  1 AS numExtracts,
+ /* End: Querying data specific to Extract operation */
+ /* Start: Querying data specific to Query operation */
+ REGEXP_CONTAINS(query.query, 'cloudaudit_googleapis_com_data_access_')
+ AS isAuditDashboardQuery,
```

Comment:
move to previous line.

Reply:
Will exceed 80 char if moved to previous line.